tests/Integration/inc/Engine/Media/AboveTheFold/AJAX/Subscriber/addLcpData.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace WP_Rocket\Tests\Integration\Inc\Engine\Media\AboveTheFold\AJAX\Subscriber;
+
+use WP_Rocket\Tests\Integration\AjaxTestCase;
+
+/**
+ * @covers WP_Rocket\Engine\Media\AboveTheFold\AJAX\Subscriber::add_lcp_data
+ *
+ * @group AboveTheFold
+ */
+class Test_AddLcpData extends AjaxTestCase {
+	private $allowed;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->action = 'rocket_lcp';
+
+		$this->clean_table();
+	}
+
+	/**
+	 * $_POST is cleared in parent method
+	 *
+	 * @return void
+	 */
+	public function tear_down() {
+		remove_filter( 'rocket_above_the_fold_optimization', [ $this, 'set_allowed' ] );
+		$this->clean_table();
+
+		parent::tear_down();
+	}
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldReturnExpected( $config, $expected ) {
+		$_POST['rocket_lcp_nonce'] = wp_create_nonce( 'rocket_lcp' );
+		$_POST['action']           = 'rocket_lcp';
```

Comment:
You set `$this->action = 'rocket_lcp';` in the `set_up()` function, but you don't use it here. Is there a reason?

Reply:
It's used in the parent class, when calling the ajaxAction